public/locations-list-map.js

```diff
@@ -278,125 +377,103 @@ window.onFilterChange = function (elem, scrollNeeded) {
-                 const open_accepted = entry["open_box"];
-                 // Convert the lat and lng fields to numbers
-                 if (!isNaN(Number(latitude))) {
-                     var marker = addMarkerToMap(map, Number(latitude), Number(longitude),
-                         address, name, instructions, accepting, open_accepted);
-                     markers.push(marker);
-                 }
-             }
-         }
-     }
-
-     if (singleStateFilter) {
-       // Center the map to a state if only one is set in query params
-       centerMapToState(map, firstState);
-     } else {
-       // Center the map on the nearest markers to the user if possible
-       centerMapToNearestMarkers(map, markers);
-     }
+
+
+/**
+ * Called once Google Maps API is fully loaded ready to run.
+ *
+ * IMPORTANT: This is exposed via window.initMap so the script can execute it.
+ */
+function initMap() {
+  var data_by_location = window.data_by_location;
+
+  var element = document.getElementById('map');
+
+  if (element == null) {
+    alert('could not find map div');
+  }
+
+  $(".map-container").show();
+
+  const singleStateFilter = showStates && showStates.length === 1;
+  const firstState = showStates[0];
+
+  // Load map at our initial center + zoom level.
+  map = new google.maps.Map(element, {zoom: initial_zoom, center: inital_center});
+
+  // filter states if there is a state filter
+  const filteredStates = Object.keys(data_by_location).filter((stateCode) => (
+    (
+      singleStateFilter
+      && firstState.toUpperCase() === stateCode.toUpperCase()
+      && stateLocationMappings[stateCode.toUpperCase()]
+      // add markers when no single state filter or if state code is invalid
+    ) || (!singleStateFilter || !stateLocationMappings[firstState.toUpperCase()])
+  ));
+
+  for (const state of filteredStates.sort()) {
+    const cities = data_by_location[state];
+
+    for (const city of Object.keys(cities).sort()) {
+      for (const entry of cities[city]) {
+        const name = entry["name"];
+        const address = entry["address"];
+        const latitude = entry["lat"];
+        const longitude = entry["lng"];
+        const instructions = entry["instructions"];
+        const accepting = entry["accepting"];
+        const open_accepted = entry["open_box"];
+        // Convert the lat and lng fields to numbers
+        if (!isNaN(Number(latitude))) {
+          var marker = addMarkerToMap(Number(latitude), Number(longitude),
+            address, name, instructions, accepting, open_accepted);
+          markers.push(marker);
+        }
+      }
+    }
+  }
+
+  if (singleStateFilter) {
+    // Center the map to a state if only one is set in query params
+    centerMapToState(firstState);
+  } else {
+    // Center the map on the nearest markers to the user if possible
+    centerMapToMarkersNearUser();
+  }
+
+
+  // Initialize the map search autocompleter.
+  autocomplete = new google.maps.places.Autocomplete(
```

Comment:
Search controls should start out hidden and be opt-in, since their presence will push the map down so it may no longer fit in embedded views (getusppee.org/give).

Reply:
@graemer @awong-dev 

What is your opinion on this particular point? If we show a map, should the map always have a search filter or should that be optional? If so, what are the criteria for what hides it? Explicit? Or only show when not selecting a state via querystring? etc.

Reply:
@drwestco added new `hide-search` parameter for now in 8751db4, so that you can at least get rid of it if desired. Right now the default it so show, hence the parameter name. Maybe best to eventually refactor our QS params to be regular booleans without the `hide-` prefixes, but that's a discussion for later. 🤔